apps/src/templates/teacherNavigation/UnitCalendar.tsx

```diff
@@ -51,17 +68,10 @@ const UnitCalendar: React.FC = () => {
       unitNameFromProgress !== unitName
     ) {
       setIsLoading(true);
-      getAuthenticityToken()
-        .then(token => {
-          return fetch(`/dashboardapi/unit_summary/${unitName}`, {
-            method: 'GET',
-            headers: {
-              'Content-Type': 'application/json',
-              'X-CSRF-Token': token,
-            },
-          });
-        })
-        .then(response => response.json())
+      HttpClient.fetchJson<UnitSummaryResponse>(
```

Comment:
Did you test that this still works. I want to make sure that the data is getting parsed correctly with this change.

Reply:
This PR is big enough, but we should probably add some unit tests for `UnitCalendar` at some point.

Reply:
> Did you test that this still works. I want to make sure that the data is getting parsed correctly with this change.

I tested this locally in the calendar, and agreed to needing to write a test.